doc/developer/design/20230223_stabilize_with_mutually_recursive.md

```diff
@@ -138,8 +138,15 @@ From the remaining 16, based on an initial analysis it seems that:
 - 4 are relatively straight-forward to fix (size estimate `M?`),
 - 12 maintain `Let`-based context and need case-by-case reasoning (marked with `L?`).
 
-I will do a second pass of those transforms as part of MaterializeInc/materialize#17914, work on some, and update my estimates once I have more experience.
-This work is going to be parallelized with [@ggevay](https://github.com/ggevay).
+For most non-trivial transforms, we have multiple solutions at our disposal:
+
+1. A _basic_ solution which only applies the transform to bindings that are not actually recursive and treats recursive bindings as an optimization barrier.
+   Transforms using this we have are marked with _solution = basic_ in the table above.
+2. An _advanced_ solution which uses abstract interpretation based on lattice theory to propagate information through `LetRec` nodes.
+   Transforms using this solution are marked with _solution = basic_ in the table above.
+   For the _basic_ transforms the _advanced_ solution is sketched in the linked issue in case we want to improve them as future work.
+3. An _advanced_ solution that we will get without changes to the actual `Transform` code if we implement MaterializeInc/materialize#18174.
+   Those are marked with _solution = basic_ and the corresponding issue as depending on MaterializeInc/materialize#18174.
```

Comment:
Is the issue number correct? https://github.com/MaterializeInc/materialize/issues/18174 seems to be `semijoin_idempotence`. Maybe you wanted to link the key inference issue (https://github.com/MaterializeInc/materialize/issues/18553)? (and also in the table)